engine/access/rpc/backend/transactions/transactions.go

```diff
@@ -242,6 +243,22 @@ func (t *Transactions) SendRawTransaction(
 func (t *Transactions) GetTransaction(ctx context.Context, txID flow.Identifier) (*flow.TransactionBody, error) {
 	tx, err := t.transactions.ByID(txID)
 	if err == nil {
+		// Make sure the transaction is from before the hardfork
+		lightCollection, err := t.collections.LightByTransactionID(txID)
+		if err != nil {
+			return nil, status.Errorf(codes.NotFound, "transaction not found in any collection")
+		}
+		_, err = t.blocks.ByCollectionID(lightCollection.ID())
+		if err != nil {
+			if errors.Is(err, badger.ErrBlockAfterHardfork) {
+				return nil, status.Errorf(codes.NotFound, "transaction is from after the hardfork")
```

Comment:
I would recommend we don't introduced distinct error path. Rather, I would suggest to use [`BeyondArchiveThreshold`](https://github.com/onflow/flow-go/blob/559651356dc5362675cffe1ae0ace46c5b4ab3a8/storage/operation/headers.go#L42-L72) as a wrapper of  `storage.ErrNotFound` (see my [PR #8265](https://github.com/onflow/flow-go/pull/8265)). So we can _add_ checks where we want without interrupting some unmodified existing logic suddenly crashing because of unexpected error. 

```suggestion
			if operation.IsBeyondArchiveThreshold(err) {
				return nil, status.Errorf(codes.NotFound, "transaction is from after the hardfork")
```